common/src/main/java/com/revenuecat/purchases/common/logWrapper.kt

```diff
@@ -2,30 +2,82 @@ package com.revenuecat.purchases.common
+var currentVerboseLogHandler: VerboseLogHandler = DefaultVerboseLogHandler()
 
 private class DefaultLogHandler : LogHandler {
     override fun d(tag: String, msg: String) {
-        Log.d(tag, msg)
+        debugLog(msg, null)
     }
 
     override fun i(tag: String, msg: String) {
-        Log.i(tag, msg)
+        infoLog(msg, null)
     }
 
     override fun w(tag: String, msg: String) {
-        Log.w(tag, msg)
+        warnLog(msg, null)
     }
 
     override fun e(tag: String, msg: String, throwable: Throwable?) {
-        if (throwable != null) {
-            Log.e(tag, msg, throwable)
-        } else {
-            Log.e(tag, msg)
+        errorLog(msg, throwable)
+    }
+}
+
+fun verboseLogHandlerToLogHandler(value: LogHandler) = object : VerboseLogHandler {
+    override fun log(
+        level: LogLevel,
+        tag: String,
+        message: String,
+        throwable: Throwable?,
+        elements: Array<StackTraceElement>?
+    ) {
+        when (level) {
+            LogLevel.DEBUG -> value.d(tag, message)
+            LogLevel.INFO -> value.i(tag, message)
+            LogLevel.WARN -> value.w(tag, message)
+            LogLevel.ERROR -> value.e(tag, message, throwable)
+            LogLevel.VERBOSE -> return // LogHandler doesn't handle verbose
+        }
+    }
+}
+
+private class DefaultVerboseLogHandler : VerboseLogHandler {
+    override fun log(
+        level: LogLevel,
+        tag: String,
+        message: String,
+        throwable: Throwable?,
+        elements: Array<StackTraceElement>?
+    ) {
+        val messageWithStacktraceIfNeeded = message + if (Config.verboseLogs) { stacktraceString(elements) } else ""
+
+        when (level) {
+            LogLevel.DEBUG -> Log.d(tag, messageWithStacktraceIfNeeded)
+            LogLevel.INFO -> Log.i(tag, messageWithStacktraceIfNeeded)
+            LogLevel.WARN -> Log.w(tag, messageWithStacktraceIfNeeded)
+            LogLevel.ERROR -> throwable?.let {
+                Log.e(tag, messageWithStacktraceIfNeeded, it)
+            } ?: Log.e(tag, messageWithStacktraceIfNeeded)
+            LogLevel.VERBOSE -> throwable?.let {
+                Log.v(tag, messageWithStacktraceIfNeeded, it)
+            } ?: Log.v(tag, messageWithStacktraceIfNeeded)
         }
     }
+
+    private fun stacktraceString(elements: Array<StackTraceElement>?) = if (elements != null) {
+        "\n${elements.joinToString("\n")}"
+    } else {
+        // Getting the stacktrace by creating a Throwable is faster and gives the correct item (this function)
+        // for the first item of the stacktrace. It prints "java.lang.Throwable\n" in the first line,
+        // so we remove that. Using currentThread().stacktrace will print this at the beginning of the stack:
+        //      dalvik.system.VMStack.getThreadStackTrace(Native Method)
+        //      java.lang.Thread.getStackTrace(Thread.java:1841)
+        "\n${Throwable().stackTraceToString().replaceFirst("java.lang.Throwable\n", "")}"
+    }
```

Comment:
This part I am not super convinced, but I couldn't find a better way to print what we want, the lines where the log was called. I actually thought printing the whole stacktrace would be helpful.

Open for feedback on this